internals/daemon/daemon_test.go

```diff
@@ -1148,3 +1148,35 @@ services:
 	c.Assert(tasks, HasLen, 1)
 	c.Check(tasks[0].Kind(), Equals, "stop")
 }
+
+func (s *daemonSuite) TestSyscallRebootDelay(c *C) {
+	waitState := 0
+	old := shutdownSyscall
+	shutdownSyscall = func() {
+		waitState = 1
+	}
+	defer func() {
+		shutdownSyscall = old
+	}()
+	syscallReboot(time.Millisecond * 25)
+	c.Assert(waitState, Equals, 0)
+	time.Sleep(time.Millisecond * 50)
```

Comment:
There are data races that `go test -race` is showing caused by writing the global vars with `time.Sleep` as the synchronization mechanism. Per your excellent work in the other PR, I think we're going to have to use proper synchronization here. That might mean we have to expose a bit more (a channel?) in the non-test code.

Reply:
Or maybe you can send on a channel inside the modified `shutdownSyscall`?

Reply:
Should look much nicer now. Let me know what you think. Sorry for waiting your time here earlier ...

Reply:
No problemo. Looks good now, except for the note about having a timeout on the waits.